cypress/integration/project/follow.spec.ts

```diff
@@ -0,0 +1,36 @@
+import * as commands from "../../support/commands";
+
+context("project following", () => {
+  const projectId =
+    "hwd1yre8dchhkyd34h9f5h7ymbnj5tkzoezb9kuthi885ecbuu5ey69cipw";
+
+  beforeEach(() => {
+    commands.resetProxyState();
+    commands.onboardUser("cloudhead");
+    cy.visit("./public/index.html");
+  });
+
+  it("follows and unfollows", () => {
+    commands.pick("Following").click();
+    commands.pick("primary-action").contains("Look for a project").click();
+    commands.pick("search-input").type(`rad:git:${projectId}`);
+    commands.pick("follow-toggle").should("contain", "Follow");
+    commands.pick("follow-toggle").click();
+    commands
+      .pick("notification")
+      .should("contain", "You’ll be notified when this project has been found");
+
+    // The “following” list doesn’t update automatically
+    cy.reload();
```

Comment:
Why does this work without an explicit reload outside of the tests?

![noreload](https://user-images.githubusercontent.com/158411/99393768-1c71c180-28de-11eb-9b7e-8a284f4ec9e9.gif)

Reply:
I don’t know. It looks to me that outside of the tests events from the peer trigger an implicit update to the screen.

Reply:
Do you observe the relevant `localPeer` events in testing?

Reply:
Ah, that makes sense, we haven't bothered to set up the SSE stuff to work within cypress. I wonder how much work that would be?

Reply:
@rudolfs Are you saying the UI doesn't establish an SSE connection when run with cypress>

Reply:
@xla correct: https://github.com/cypress-io/cypress/issues/2747.

Reply:
I added a debug log to `localPeer.ts`:
```diff
diff --git a/ui/src/localPeer.ts b/ui/src/localPeer.ts
index ae3ca89c..9b4c2fc6 100644
--- a/ui/src/localPeer.ts
+++ b/ui/src/localPeer.ts
@@ -83,6 +83,7 @@ session.session.subscribe(sess => {
         { withCredentials: true }
       );
       eventSource.addEventListener("message", msg => {
+        console.log(msg);
         const event: Event = JSON.parse(msg.data);
         eventStore.set(event);
       });
```

And it seems like we only get the first event and no subsequent events through the `EventSource` when running within cypress. Might be worth investigating: https://github.com/cypress-io/cypress/issues?q=sse.

Reply:
To avoid any red herrings, we are not using SSE from cypress itself, so we should not be affected by the conversations of having sse support in cypress itself.

Reply:
@xla we are running the our app within the Electron shell of Cypress.

Reply:
It has nothing to do with cypress. I started the app with `proxy --test` and had the same issue.

Reply:
Created an issue: https://github.com/radicle-dev/radicle-upstream/issues/1279

Reply:
🙏